src/Services/QueuedJobService.php

```diff
@@ -1363,6 +1429,29 @@ public function isMaintenanceLockActive()
         return file_exists($path);
     }
 
+    /**
+     * Get expiry time for a worker to be operating on a job, helps to identify jobs
+     * that have stalled more accurately.
+     *
+     * @return string
+     * @throws Exception
+     */
+    protected function getWorkerExpiry(): string
+    {
+        $now = DBDatetime::now()->Rfc2822();
+        $time = new DateTime($now);
+        $timeToLive = $this->config()->get('worker_ttl');
+
+        if ($timeToLive) {
+            $time->add(new DateInterval($timeToLive));
```

Comment:
You can actually do this with only DBDatetime ([reference](https://github.com/silverstripe/silverstripe-framework/pull/9105/files)), example:

```php
$expiry = DBDatetime::now();
$timeToLive = $this->config()->get('worker_ttl'); // example: "+5 minutes"
if ($timeToLive) {
    $expiry->modify($timeToLive);
}
return $expiry->Rfc2822();
```

Reply:
That's good to know, thanks :)